php/settings/class-setting.php

```diff
@@ -881,13 +878,44 @@ public function create_setting( $slug, $params = array(), $parent = null ) {
 	 * @return Setting
 	 */
 	public function add_setting( $setting ) {
+
 		$setting->set_parent( $this );
-		$this->settings[ $setting->get_slug() ] = $setting;
-		$this->sort_settings();
+
+		// Get the position in which to insert the new setting.
+		$index = $this->get_position_index( $setting->get_param( 'priority', 10 ) );
+
+		$new_setting = array(
+			$setting->get_slug() => $setting,
+		);
+
+		// Add the new setting at the index based on the priority position.
```

Comment:
@DavidCramer 

I'm approving this PR as is, but I'd like us to add here a note explaining the issues we had with PHP 5.6 and `uasort`.